sys/drv/bcm2835_emmc.c

```diff
@@ -65,11 +67,25 @@ static emmc_error_t bcemmc_decode_errors(uint32_t interrupts) {
     return e;
 
   if (interrupts & INT_CTO_ERR)
-    e |= EMMC_ERROR_CMD_TIMEOUT;
+    e |= EMMC_ERROR_TIMEOUT;
 
   return e;
 }
 
+static inline int bcmemmc_invalid_state(bcmemmc_state_t *state) {
+  return state->errors & (~state->ignore_errors | EMMC_ERROR_INTERNAL |
+                          EMMC_ERROR_INVALID_STATE);
+}
+
+static inline emmc_error_t bcmemmc_set_error(bcmemmc_state_t *state,
+                                             emmc_error_t error_flags) {
+  state->errors = error_flags;
+  if ((error_flags & !state->ignore_errors) && !bcmemmc_invalid_state(state)) {
```

Comment:
It doesn't make sense to me. First, I think you wanted to do bitwise negation instead of logical negation in the first expression (i.e. `!state->ignore_errors` -> `~state->ignore_errors`). If my suggestion is valid then the second expression is useless. It is equivalent to the following:
`(state->errors & ~state->ignore_errors) && !(state->errors & (~state->ignore_errors | <something>))`
which can be written as
`(error & ~ignore) && !((error & ~ignore) | (error & <something>))`
further
`(error & ~ignore) && !(error & ~ignore) && !(error & <something>)`
this yields
`false`

Reply:
Fixed and simplified this logic.